blaze/expr/expressions.py

```diff
@@ -104,8 +101,7 @@ class Expr(Node):
     contains shared logic and syntax.  It in turn inherits from ``Node`` which
     holds all tree traversal logic
     """
-
-    __slots__ = '_hash', '__weakref__'
+    __slots__ = '_hash', '__weakref__', '__dict__'
```

Comment:
We will now allocate one extra dictionary per expression. We will still preallocate the space for all of the other slots so there is still some use for the `__slots__`.